site/content/how-to/data-plane-configuration.md

```diff
@@ -157,3 +157,45 @@ as arguments and add `/bin/sh` as the command. The deployment manifest should lo
   - /bin/sh
 ...
 ```
+
+## Configure PROXY protocol and RewriteClientIP settings
+
+When the request is passed through multiple proxies or load balancers, the client IP is set to the IP address of the server that last handled the request. To preserve the original client IP address, you can configure `RewriteClientIP` settings in the `NginxProxy` resource. `RewriteClientIP` has the fields: _mode_, _trustedAddresses_ and _setIPRecursively_.
+
+**Mode** determines how the original client IP is passed through multiple proxies and the way load balancer is set to receive it. It can have two values:
+
+  1. `ProxyProtocol` is a protocol that carries connection information from the source requesting the connection to the destination for which the connection was requested.
+  2. `XForwardedFor` is a multi-value HTTP header that is used by proxies to append IP addresses of the hosts that passed the request.
+
+The choice of mode depends on how the load balancer fronting NGINX Gateway Fabric receives information.
+
+**TrustedAddresses** are used to specify the IP addresses of the trusted proxies that pass the request. These can be in the form of CIDRs, IPs, or hostnames. For example, if a load balancer is forwarding the request to NGINX Gateway Fabric, the IP address of the load balancer should be specified in the `trustedAddresses` list to inform NGINX that the forwarded request is from a known source.
+
+**SetIPRecursively** is a boolean field used to enable recursive search when selecting the client's address from a multi-value header. It is applicable in cases where we have a multi-value header containing client IPs to select from, i.e., when using `XForwardedFor` mode.
+
+The following command creates an `NginxProxy` resource with `RewriteClientIP` settings that set the mode to XForwardedFor, enables recursive search for finding the client IP and sets a CIDR, IPAddress and Hostname in the list of trusted addresses to find the original client IP address.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxProxy
+metadata:
+  name: ngf-proxy-config
+spec:
+  config:
+    rewriteClientIP:
+      mode: XForwardedFor
```

Comment:
This example shows for the "XForwardedFor" Mode, Do we need to consider the ProxyProtocol yaml example as well?

Reply:
might be redundant to add another example. We are mentioning that there are two possible modes that they could use, this is just an example of how to use the fields.

Reply:
How are we communicating to the user which fields are required for ProxyProtocol mode? The fields that need to be configured differ between ProxyProtocol and XForwardedFor modes.

Reply:
There is no harm if `setIPRecursively` is set when mode is `ProxyProtocol` so i don't think we need to make a special note or extra example. 

But since the doc starts with `ProxyProtocol` and which is the more important use case, I can use that mode for the example.